### ISSUE TITLE
">" Character Caused Verification to Fail

### ISSUE THREAD

**Issue by willysr** [kylehuff/webpg-firefox#11]
Special character such as > used for replying caused verification to fail
i tested using no ">" character and it's success, but when i add ">" as part of the message (not important where they are placed on the text), it will fail to verify, be on new message or replying message

**Comment by willysr** [comment]
Still not yet fixed (at least not in Chrome)
i haven't tested on Firefox as i'm in the middle of downloading something, but i will post an update when it's done
![chrome](https://f.cloud.github.com/assets/240100/493203/5eaa8126-bb28-11e2-8315-324ddbbeabb5.jpg)

**Comment by kylehuff** [comment]
@willysr would you be willing to pull down the latest version of webpg-chrome (master) and give it a go?

Some changes to note -
- WebPG now supports sending and parsing PGP/MIME messages
- Issues with validating should (mostly) be a thing of the past.

Some cases where issues will arise is with special characters in the body of the email, as the quoted-printable decoder is not working as it should...

**Comment by willysr** [comment]
Sure.
I'm assuming it should be fixed in Firefox as well?

**Comment by kylehuff** [comment]
Yes, I only mentioned the chrome repo because it is the easiest (IMO) to install from a repository.

Both of the repositories are synced though.

**Comment by willysr** [comment]
Hm... strange
firefox works, but Google Chrome plugin installation failed

We encountered a problem while attempting to load WebPG

Error (-1): WebPG Plugin failed to load

Suggestions:

The plugin was unable to load

Error Details

Error in Method: undefined
Error Code: -1
Error String: WebPG Plugin failed to load
File: webpgPluginAPI.cpp
Line: -1
System Information

Platform: Linux i686
App Version: 5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.114 Safari/537.36
User Agent: Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.114 Safari/537.36
Product: Gecko
Sub-product: 20030107
Vendor: Google Inc.
Language: en-US

**Comment by willysr** [comment]
Tested on Firefox:

I'm sending this email

hi willysr@slackware-id.org

try the new css body > p

## 

Willy Sudiarto Raharjo
Personal Blog : http://willysr.blogspot.com
Linux Blog: http://slackblogs.blogspot.com

and here's what i got

![webpg](https://cloud.githubusercontent.com/assets/240100/3086543/ea3cde0a-e539-11e3-9f77-66eb4fb19fb3.png)

**Comment by kylehuff** [comment]
That looks about right (the email was encoded with quoted-printable, however the quoted-printable decoder is disabled in WebPG).

As you can see, it validates, but the display is slightly garbled.

Regarding the issue in chrome, that seems very strange. If it loads in Firefox on the same machine, it should load without incident in chrome...

**Comment by willysr** [comment]
I don't know. I'm using the latest Google Chrome 35 here

**Comment by kylehuff** [comment]
Yeah, I just installed 35, and I have the same result... I know they are supposed to be dropping support for NPAPI, I just didn't think it was right now. I'll investigate.

**Comment by willysr** [comment]
I also found a new strange bug
i received a signed message, in which i don't have the sender's public key, but then webPG convert it into a scrambled text just like it was an encrypted text

![webpg](https://cloud.githubusercontent.com/assets/240100/3086634/8c832a72-e53d-11e3-94d4-4a5ffccee7e5.png)

**Comment by kylehuff** [comment]
If I had to guess, that is a base64 encoded email... If it is, once I fix the encode/decoder(s) that will go away.

**Comment by willysr** [comment]
Yep
looking at the headers

Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: base64

**Comment by kylehuff** [comment]
Yes, the issue with chrome is due to the discontinuation of NPAPI support.

I have created an issue to address this problem kylehuff/webpg-chrome#102

**Comment by kylehuff** [comment]
I re-enabled the QP and Base64 decoders - the QP decoder still doesn't work well, however, it is better than not having any decoding at all (most for Base64).
